### PR TITLE
Create plot: melting of a body having a high velocity

### PR DESCRIPTION
**Pull request by Voldoned**

### CODE REVIEW

File: plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+from symplyphysics.laws.thermodynamics import thermal_energy_from_mass_and_temperature as energy_heating_law
+from symplyphysics.laws.thermodynamics import energy_to_melt_from_mass as energy_melting_law
+from symplyphysics.laws.conservation import mechanical_energy_after_equals_to_mechanical_energy_before as energy_conservation_law
+from symplyphysics.laws.dynamics import kinetic_energy_from_mass_and_velocity as kinetic_energy_law
+
+matters_parametrs = {
+    "Fe": {
+        "specific_heat_heating": 460,
+        "specific_heat_melting": 270_000,
+        "temperature_of_melting": to_kelvin(Celsius(1_400)),
```
Comment (blackyblack) [inline plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py:17]: You can write just numbers and convert later.

File: plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+from sympy.plotting import plot
+from sympy.plotting.plot import MatplotlibBackend
+from symplyphysics import print_expression
+from symplyphysics.core.symbols.celsius import to_kelvin, Celsius
+from symplyphysics.laws.thermodynamics import thermal_energy_from_mass_and_temperature as energy_heating_law
+from symplyphysics.laws.thermodynamics import energy_to_melt_from_mass as energy_melting_law
+from symplyphysics.laws.conservation import mechanical_energy_after_equals_to_mechanical_energy_before as energy_conservation_law
+from symplyphysics.laws.dynamics import kinetic_energy_from_mass_and_velocity as kinetic_energy_law
+
+matters_parametrs = {
```
Comment (blackyblack) [inline plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py:13]:
```suggestion
matter_parameters = {
```

File: plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+print(f"Total equation:\n{print_expression(answer)}")
+
+koefficient_of_melting_meteorite_to_plots = answer.subs({
+    koefficient_of_transfer_to_kinetic: 0.80,   # percents
+    temperature_of_meteorite: 300   # kelvins
+})
+
+base_plot = plot(title="The proportion of a molten meteorite depending on its velocity",
+    xlabel=r"$v, m/s$",
+    ylabel=r"$m_{melt-meteorite} / m_{meteorite}$",
```
Comment (blackyblack) [inline plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py:97]: Does not look good on the plot. Maybe just `molten_ratio`?

File: plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    backend=MatplotlibBackend,
+    legend=True,
+    show=False)
+
+masses_ratio_supremum = 1
+masses_ratio_minimum = 0
```
Comment (blackyblack) [inline plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py:103]: There is convention to write constants in all caps, eg `MASSES_RATIO_SUPREMUM`
BTW why supremum, not `maximum`?

File: plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+for matter in matters_parametrs:
```
Comment (blackyblack) [inline plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py:104]: Use `for matter, parameters in matters_parametrs.items()`

**Review by blackyblack — APPROVED** [review]

File: plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    ylabel=r"$molten ratio$",
+    backend=MatplotlibBackend,
+    legend=True,
+    show=False)
+
+MASSES_RATIO_MAXIMUM = 1
+MASSES_RATIO_MINIMUM = 0
+for matter, parameters in matters_parametrs.items():
+    masses_ratio_to_subplot = koefficient_of_melting_meteorite_to_plots.subs({
+        specific_heat_heating_meteorite: parameters["specific_heat_heating"],   # joules / (kilogram * kelvin)
```
Comment (blackyblack) [inline plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py:106]: MyPy is complaining about types here. Could you fix it?

File: plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py (hunk 2 above)
Comment (blackyblack) [reply plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py:13]: Not fixed

Comment (blackyblack) [reply plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py]: Don't you see typo here?

File: plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py (hunk 6 above)
Comment (Voldoned) [reply plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py:106]: Convert to `float`?

Comment (blackyblack) [reply plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py:106]: Nope. I will show how to do it.

**Pull request merged**